data/mods/Magiclysm/achievements/statistics.json

```
@@ -0,0 +1,54 @@
+[
+  {
+    "id": "avatar_levels_spell",
+    "type": "event_transformation",
+    "event_type": "character_levels_spell",
+    "value_constraints": { "character": { "equals_statistic": "avatar_id" } }
+  },
+  {
+    "id": "avatar_learns_spell",
+    "type": "event_transformation",
+    "event_type": "character_learns_spell",
+    "value_constraints": { "character": { "equals_statistic": "avatar_id" } },
+    "drop_fields": [ "spell" ]
+  },
+  {
+    "id": "num_spells_learned",
+    "type": "event_statistic",
+    "stat_type": "count",
+    "event_transformation": "avatar_learns_spell",
+    "description": { "str": "learned a spell", "str_pl": "number of spells learned" }
+  },
+  {
+    "id": "avatar_forgets_spell",
+    "type": "event_transformation",
+    "event_type": "character_forgets_spell",
+    "value_constraints": { "character": { "equals_statistic": "avatar_id" } },
+    "drop_fields": [ "spell" ]
```

```suggestion
    "drop_fields": [ "character" ]
```
You might care about the `spell` field, but you certainly won't care about the `character` one, since it's always the avatar.  I believe your other stats should still work unchanged when the `spell` field is there.
(Similarly for the others).